src/incubator/Slider/index.tsx

```diff
@@ -0,0 +1,390 @@
+  const defaultThumbStart = useSharedValue({x: 0, y: 0});
+  const isPressedRange = useSharedValue(false);
+  const rangeThumbOffset = useSharedValue({x: 0, y: 0});
+  const rangeThumbStart = useSharedValue({x: 0, y: 0});
+
+  const defaultThumbStyle: StyleProp<ViewStyle> = [
+    styles.thumb,
+    {backgroundColor: disabled ? Colors.$backgroundDisabled : thumbTintColor || Colors.$backgroundPrimaryHeavy}
+  ];
+  const _thumbStyle = useSharedValue(JSON.parse(JSON.stringify(StyleSheet.flatten(thumbStyle || defaultThumbStyle))));
+  const _activeThumbStyle = 
+    useSharedValue(activeThumbStyle ? JSON.parse(JSON.stringify(StyleSheet.flatten(activeThumbStyle))) : undefined);
+
+  const trackAnimatedStyles = useAnimatedStyle(() => {
+    if (useRange) {
+      return {
+        transform: [{translateX: defaultThumbOffset.value.x * rtlFix}],
+        width: activeTrackWidth.value
+      };
+    } else {
+      return {
+        width: activeTrackWidth.value
+      };
+    }
+  });
+  
+  const defaultThumbAnimatedStyles = useAnimatedStyle(() => {
+    const activeStyle = isPressedDefault.value ? _activeThumbStyle.value : _thumbStyle.value;
+    return {
+      transform: [
+        {translateX: (defaultThumbOffset.value.x - thumbSize.value.width / 2) * rtlFix},
+        {scale: withSpring(!disableActiveStyling && isPressedDefault.value ? 1.3 : 1)}
+      ],
+      ...activeStyle
+    };
+  });
+
+  const rangeThumbAnimatedStyles = useAnimatedStyle(() => {
+    const activeStyle = isPressedDefault.value ? _activeThumbStyle.value : _thumbStyle.value;
+    return {
+      transform: [
+        {translateX: (rangeThumbOffset.value.x - thumbSize.value.width / 2) * rtlFix},
+        {scale: withSpring(!disableActiveStyling && isPressedRange.value ? 1.3 : 1)}
+      ],
+      ...activeStyle
+    };
+  });
+
+  const updateDefaultThumb = (x: number) => {
+    'worklet';
+    defaultThumbOffset.value = {x, y: 0};
+    defaultThumbStart.value = {x, y: 0};
+    activeTrackWidth.value = Math.abs(useRange ? rangeThumbStart.value.x - x : x);
+    runOnJS(onChange)(useRange ? {min: x, max: rangeThumbStart.value.x} : x);
+  };
+
+  const updateRangeThumb = (x: number) => {
+    'worklet';
+    rangeThumbOffset.value = {x, y: 0};
+    rangeThumbStart.value = {x, y: 0};
+    activeTrackWidth.value = x - defaultThumbStart.value.x;
+    runOnJS(onChange)({min: defaultThumbStart.value.x, max: x});
+  };
+
+  const defaultThumbGesture = Gesture.Pan()
+    .onBegin(() => {
+      isPressedDefault.value = true;
+      activeThumb.value = ThumbType.DEFAULT;
+    })
+    .onUpdate(e => {
+      onSeekStart?.();
+      let newX = defaultThumbStart.value.x + e.translationX * (shouldDisableRTL ? 1 : rtlFix);
+      if (newX < 0) { // bottom edge
+        newX = 0;
+      } else if (!useRange && newX > trackSize.value.width) { // top edge
+        newX = trackSize.value.width;
+      }
+      if (newX <= rangeThumbStart.value.x - rangeGap && newX >= 0) { // range
```

Comment:
I found this part a little difficult to understand
Why are you comparing against the rangeSlider value without first checking if useRange is even enabled?
Also, not sure I understand how activeTrackWidth works.. can you elaborate on this part

Reply:
'range' here in the comment refers to a range of the x value, not range Thumb i.e. when the x is in the range of 0 and trackWidth.
activeTrackWidth is the width of the "colored" track, it goes from 0 to x on a default slider and between the default thumb position and range thumb position in the range slider. When the thumbs are moving I update it's width to flow the thumbs